rs/nns/governance/src/proposals/install_code.rs

```diff
@@ -299,6 +349,38 @@ mod tests {
         );
     }
 
+    #[cfg(feature = "test")]
+    #[test]
+    fn test_upgrade_root_protocol_canister() {
+        let install_code = InstallCode {
+            canister_id: Some(ROOT_CANISTER_ID.get()),
+            wasm_module: Some(vec![1, 2, 3]),
+            install_mode: Some(CanisterInstallMode::Upgrade as i32),
+            arg: Some(vec![4, 5, 6]),
```

Comment:
NC: Future work?

We could maybe check that this is Candid-encoded. We don't necessary have to read the whole thing. I think Candid has some magic bytes at the beginning.

Reply:
I think in most of the cases we can but not sure if we can say it's always the case. Ironically, NNS Governance init args is protobuf-encoded...